packages/realm/src/app-services/User.ts

```diff
@@ -58,7 +61,25 @@ export interface UserIdentity {
   providerType: ProviderType;
 }
 
+/** @internal */
+function cleanArguments(args: unknown[] | unknown): unknown[] | unknown {
+  if (Array.isArray(args)) {
+    return args.map((x) => cleanArguments(x));
+  } else if (typeof args === "object") {
+    const result: { [key: string]: unknown } = {};
+    for (const [k, v] of Object.entries(args as object)) {
```

Comment:
You shouldn't need the `as object` cast. My best guess is that this is because `typeof null == "object"`, so we are entering this block incorrectly for null args. You probably want to make the `if` something like `if (args !== null and typeof args === "object")`.

Or even better, use an `if (args  === null || typeof args != "object") return args` early return.